src/NUnitEngine/nunit.engine.tests/Drivers/NUnit3FrameworkDriverTests.cs

```diff
@@ -149,11 +147,23 @@ public void RunTestsAction_AfterLoad_ReturnsRunnableSuite()
         public void RunTestsAction_WithoutLoad_ThrowsInvalidOperationException()
         {
             var ex = Assert.Catch(() => _driver.Run(new NullListener(), TestFilter.Empty.Text));
-            if (ex is System.Reflection.TargetInvocationException)
+            if (ex is TargetInvocationException)
                 ex = ex.InnerException;
             Assert.That(ex, Is.TypeOf<InvalidOperationException>());
             Assert.That(ex.Message, Is.EqualTo(LOAD_MESSAGE));
         }
+
+        [Test]
+        public void RunTestsAction_WithInvalidFilterElement_ThrowsNUnitEngineException()
+        {
+            _driver.Load(_mockAssemblyPath, _settings);
+
+            var invalidFilter = "<filter><invalidElement>foo</invalidElement></filter>";
+            var ex = Assert.Catch(() => _driver.Run(new NullListener(), invalidFilter));
+            if (ex is TargetInvocationException)
```

Comment:
A really good question! Since the driver may be user-written or third-party, it could have any kinds of errors in it. If it does throw an unexpected exception, we should be wrapping that in an NUnitEngineException that basically says "the driver did something strange - here's the exception it through" Now that I think about it, we should probably wrap any exception from the driver __except__ an NUnitEngineException, although that doesn't have to be done for this PR unless you want to.

So, what about this case? The driver knows it's talking to the nunit 3 framework. The framework returns an exception, which will almost always be a TargetInvocationException, since we are using reflection. It doesn't know the cause of that exception - it could be any kind of error in the framework. So I guess it should just wrap the exception (inner exception in the case of TIE) in an NUnitEngineException. Does that make sense?